source/adapters/opencl/sampler.cpp

```diff
@@ -159,6 +159,12 @@ urSamplerGetInfo(ur_sampler_handle_t hSampler, ur_sampler_info_t propName,
                 sizeof(ur_sampler_addressing_mode_t));
 
   size_t CheckPropSize = 0;
+  // ur_bool_t have a size of uint8_t, but cl_bool size have the size of
+  // uint32_t so this makes a mapping between them.
+  if (propName == UR_SAMPLER_INFO_NORMALIZED_COORDS &&
+      propSize == sizeof(ur_bool_t)) {
+    propSize = sizeof(cl_bool);
```

Comment:
I think we want an entirely different path for `INFO_NORMALIZED_COORDS`, this is passing `pPropValue` directly into `clGetSamplerInfo` and returning the property size we get from CL directly in `pPropSizeRet`. We should really be passing a `cl_bool*` to the CL api and converting it to `ur_bool_t` to be returned, as well as returning `sizeof(ur_bool_t)` for `pPropSizeRet`

Reply:
Good point, I changed that. however, not sure if I handled it in the most efficient way (redundant code I mean) so let me know if I could optimize this code more.

Reply:
nah looks good, we have similar code to deal with edge cases elsewhere in the CL adapter